### PR TITLE
PER-9196

### PR DESCRIPTION
**Pull request by liam-lloyd**
Using the new AWS clients required node 14, so this is built atop @Fenn-CS's Node 14 branch

### CODE REVIEW

**Comment by cecilia-donnelly** [comment]
Can you rebase when you get a chance?

**Review by cecilia-donnelly — COMMENTED** [review]
Okay! Thanks for your patience on this. Thank you so much for adding API documentation! What a joy.

So it looks like the previous bug in the AWS SDK has been resolved and we don't have to roll our own `createPresignedPost` anymore, is that correct? Ideally (for me) that change would be a separate commit since it seems unrelated to the multi part upload work - correct me if I'm wrong! No need to change it now, just commenting on it to make sure I'm not misunderstanding.

Going to look at the backend counterpart now and test the whole thing.

**Comment by liam-lloyd** [comment]
> So it looks like the previous bug in the AWS SDK has been resolved and we don't have to roll our own createPresignedPost anymore, is that correct? 

Basically, yes. This commit updates to `upload-service` to a newer major version of the AWS SDK, and it makes our `createPresignedPost` function unnecessary

> Ideally (for me) that change would be a separate commit since it seems unrelated to the multi part upload work - correct me if I'm wrong! No need to change it now, just commenting on it to make sure I'm not misunderstanding.

You're right, that would have been a better way to organize this PR

File: src/controllers/fileDestinationUrl.controller.ts [hunk 1]
```diff
@@ -1,29 +1,122 @@
...
+  try {
+    validateCreateFileDestinationUrlParams(req.body);
+  } catch (err) {
+    if (isValidationError(err)) {
+      res.status(400).json({ error: err });
+      return;
+    }
+    next(err);
+  }
+  if (validateCreateFileDestinationUrlParams(req.body)) {
```
[the remark this answers is not in the document]
Comment (liam-lloyd) [reply src/controllers/fileDestinationUrl.controller.ts:27]: It is a weird thing I'm doing here, and I thought Typescript was forcing me to do it, but I've just discovered a better approach. Update incoming.

**Review by cecilia-donnelly — APPROVED** [review]
Thanks for the documentation here as well [oops already said that]. It all seems reasonable! I always feel like I'm missing something when I don't have any questions/concerns about bigger PRs, but I don't see anything at the moment. :)

**Pull request merged**